storage/client/file/src/main/java/com/azure/storage/file/implementation/FilesImpl.java

```diff
@@ -228,8 +231,9 @@ public FilesImpl(AzureFileStorageImpl client) {
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @return a Mono which performs the network request upon subscription.
      */
-    public Mono<FilesDownloadResponse> downloadWithRestResponseAsync(String shareName, String filePath, Integer timeout, String range, Boolean rangeGetContentMD5, Context context) {
-        return service.download(this.client.url(), shareName, filePath, timeout, this.client.version(), range, rangeGetContentMD5, context);
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<FilesDownloadResponse> downloadWithRestResponseAsync(Integer timeout, String range, Boolean rangeGetContentMD5, Context context) {
```

Comment:
Why the shareName and filePath removed? How these two para passed into the service request?